### PR TITLE
Add methods to render bem, add html str to report

### PR DESCRIPTION
**Pull request by teonlamont**
Addresses #1793.

### CODE REVIEW

File: mne/report.py [hunk 1]
```diff
@@ -839,6 +840,51 @@ def add_images_to_section(self, fnames, captions, scale=None,
...
+        for html, caption in zip(htmls, captions):
+            caption = 'custom plot' if caption == '' else caption
+            sectionvar = self._sectionvars[section]
+
+            self.fnames.append('%s-#-%s-#-custom' % (caption, sectionvar))
+            self._sectionlabels.append(sectionvar)
+            self.html.append(html)
+
+    def render_bem(self, subject, subjects_dir, decim, n_jobs):
+        """Render an bem slider html str.
```
Comment (Eric89GXL) [inline mne/report.py]: s/an/a

File: mne/report.py [hunk 2]
```diff
@@ -839,6 +840,51 @@ def add_images_to_section(self, fnames, captions, scale=None,
...
+
+        Parameters
+        ----------
+        subject : str
+            Subject name.
+        subjects_dir : str | None
+            Path to the SUBJECTS_DIR. If None, the path is obtained by using
+            the environment variable SUBJECTS_DIR.
```
Comment (Eric89GXL) [inline mne/report.py]: `subjects_dir` is usually moved toward the end since it's less likely to be necessary / useful than other arguments (e.g., `decim`)

**Comment by Eric89GXL** [comment]
It would be good to have a test at least for the HTML-adding functionality. The BEM would be nice, too, but it's only a one-line wrapper so probably overkill.

File: mne/report.py (hunk 1 above)
Comment (agramfort) [inline mne/report.py]: Please add default parameters except for subject

File: mne/report.py (hunk 2 above)
Comment (Eric89GXL) [inline mne/report.py]: You updated the parameter order above, but not in the docstring

**Comment by teonlamont** [comment]
test added and travis et al are happy again

**Comment by Eric89GXL** [comment]
Nice, thanks!

File: mne/tests/test_report.py [hunk 3]
```diff
@@ -196,4 +197,19 @@ def test_render_mri_without_bem():
...
+    """Test adding html str to mne report.
+    """
+    report = Report(info_fname=raw_fname,
+                    subject='sample', subjects_dir=subjects_dir)
+    html = report.render_bem('sample', subjects_dir=subjects_dir)
+    caption, section = 'html', 'html_section'
+    report.add_htmls_to_section(html, caption, section)
+    idx = report._sectionlabels.index(section)
+    html_compare = report.html[idx]
+    assert_equal(html, html_compare)
```
Comment (agramfort) [inline mne/tests/test_report.py]: How Long does it take to run? Maybe use decim bigger than 2

File: doc/source/whats_new.rst [hunk 4]
```diff
@@ -77,6 +77,7 @@ Changelog
 
    - Add support for Savitsky-Golay filtering of Evoked and Epochs by `Eric Larson`_
 
+   - Add `render_bem`, add_htmls_to_section` methods to `mne.Report` by `Teon Brooks`_
```
Comment (dengemann) [inline doc/source/whats_new.rst]: one ` was missing

**Pull request merged**